backend/Step_Time_Calculation.py

```diff
@@ -0,0 +1,29 @@
+import numpy as np
+
+def calculate_step_time(force_data, threshold=20.0):
+    """
+    Calculate step time from force data collected from a Bertec treadmill.
+
+    Parameters:
+    force_data (list): A list of tuples where each tuple contains (time, force).
+    threshold (float): The force threshold to determine heel strikes and toe-offs.
+
+    Returns:
+    list: A list of calculated step times in seconds
+    """
+    step_times = []
+    step_start_time = None
+
+    for time, force in force_data:
+        # Detect heel strike
+        if force >= threshold and step_start_time is None:
+            step_start_time = time  # Start timing for a step
+
+        # Detect toe-off
+        elif force < threshold and step_start_time is not None:
+            step_time = time - step_start_time  # Calculate step duration
+            step_times.append(step_time)
+            step_start_time = None  # Reset for the next step
+            
+    return step_times
```

Comment:
Your tests are failing. I believe your previous implementation with moving averages had accurate calculations. This implementation does not account for the first step time. Because there's no accountability for the first step time being alone, the program always calculates the MA of it, which will always be inaccurate. The first step time should be returned as is. if you divide it by 2 it'd be too far from the actual value.
![image](https://github.com/user-attachments/assets/4a367e0b-b65e-412c-b666-59060efb0b28)


This part of your previous implementation that is missing is what causes this miscalculation
![image](https://github.com/user-attachments/assets/785bc914-d99f-4ea6-be67-321dee7e5873)


All you need to do to make the previous implementation better was to return only one value (step time), and return it as a digit, not a formatted string.